internal/services/appservice/static_web_app_link_backend.go

```diff
@@ -0,0 +1,241 @@
+		"static_web_app_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: staticsites.ValidateStaticSiteID,
+		},
+
+		"backend_resource_id": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.Any(
+				apimanagementservice.ValidateServiceID,
+				commonids.ValidateAppServiceID,
+				containerapps.ValidateContainerAppID,
+			),
+		},
+	}
+}
+
+func (r StaticWebAppLinkBackendResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"id": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+	}
+}
+
+func (r StaticWebAppLinkBackendResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.AppService.StaticSitesClient
+			model := StaticWebAppLinkBackendModel{}
+
+			if err := metadata.Decode(&model); err != nil {
+				return err
+			}
+
+			staticAppId, err := staticsites.ParseStaticSiteID(model.StaticWebAppID)
+			if err != nil {
+				return err
+			}
+
+			loc := ""
+			linkedBackend := linkedBackend{}
+
+			if apiManagementId, err := apimanagementservice.ParseServiceID(model.BackendResourceId); err == nil {
+				linkedBackend.id = apiManagementId.ID()
+				linkedBackend.name = apiManagementId.ServiceName
+
+				backendClient := metadata.Client.ApiManagement.ServiceClient
+
+				apiManagement, err := backendClient.Get(ctx, *apiManagementId)
+				if err != nil {
+					return fmt.Errorf("reading specified %s: %+v", *apiManagementId, err)
+				}
+
+				if appModel := apiManagement.Model; appModel != nil {
+					loc = location.Normalize(appModel.Location)
+				}
+
+			} else if appServiceId, err := commonids.ParseAppServiceID(model.BackendResourceId); err == nil {
+				linkedBackend.id = appServiceId.ID()
+				linkedBackend.name = appServiceId.SiteName
+
+				backendClient := metadata.Client.AppService.WebAppsClient
+
+				appService, err := backendClient.Get(ctx, *appServiceId)
+				if err != nil {
+					return fmt.Errorf("reading specified %s: %+v", *appServiceId, err)
+				}
+
+				if appModel := appService.Model; appModel != nil {
+					loc = location.Normalize(appModel.Location)
+				}
+
+			} else if containerAppId, err := containerapps.ParseContainerAppID(model.BackendResourceId); err == nil {
+				linkedBackend.id = containerAppId.ID()
+				linkedBackend.name = containerAppId.ContainerAppName
+
+				backendClient := metadata.Client.ContainerApps.ContainerAppClient
+
+				containerApp, err := backendClient.Get(ctx, *containerAppId)
+				if err != nil {
+					return fmt.Errorf("reading specified %s: %+v", *containerAppId, err)
+				}
+
+				if appModel := containerApp.Model; appModel != nil {
+					loc = location.Normalize(appModel.Location)
+				}
+
+			} else {
+				return fmt.Errorf("unsupported backend resource type")
+			}
```

Comment:
Can we update this to a `switch` using the validate functions? Since the Schema validation gates the values here before we get here, we shouldn't need the error/defaul case.